src/layouts/components/Homepage/HeroBody.tsx

```diff
@@ -93,50 +364,41 @@ export const HeroBody = ({
             </Text>
           </Flex>
         </FormControl>
-        <FormControl isRequired isInvalid={!!errors.title}>
-          <FormLabel>Hero title</FormLabel>
-          <Input
-            placeholder="Hero title"
-            id={`section-${index}-hero-title`}
-            value={title}
-            onChange={onChange}
-          />
-          <FormErrorMessage>{errors.title}</FormErrorMessage>
-        </FormControl>
-        <FormControl isRequired isInvalid={!!errors.subtitle}>
-          <FormLabel>Hero subtitle</FormLabel>
-          <Input
-            placeholder="Hero subtitle"
-            id={`section-${index}-hero-subtitle`}
-            value={subtitle}
-            onChange={onChange}
-          />
-          <FormErrorMessage>{errors.subtitle}</FormErrorMessage>
-        </FormControl>
-        <Box w="100%">
-          {/* TODO: migrate this to design system components */}
-          <FormContext
-            hasError={!!errors.background}
-            onFieldChange={onChange}
-            isRequired
-          >
-            <Box mb="0.75rem">
-              <FormTitle>Hero background image</FormTitle>
-            </Box>
-            <FormFieldMedia
-              value={background}
-              id={`section-${index}-hero-background`}
-              inlineButtonText="Select"
-            />
-            <FormError>{errors.background}</FormError>
-          </FormContext>
-        </Box>
+        <Divider my="0.25rem" />
+        <HeroLayoutForm>
+          {({ currentSelectedOption }) => {
+            if (currentSelectedOption === HERO_LAYOUTS.CENTERED.value) {
+              return <HeroCenteredLayout {...rest} />
+            }
+
+            if (currentSelectedOption === HERO_LAYOUTS.IMAGE_ONLY.value) {
+              return <HeroImageOnlyLayout {...rest} />
+            }
+
+            if (currentSelectedOption === HERO_LAYOUTS.SIDE_SECTION.value) {
+              return (
+                <HeroSideSectionLayout
+                  {...rest}
+                  size="half"
+                  alignment="left"
+                  backgroundColor="black"
```

Comment:
This resets the values every time the user selects a different option then comes back - are we okay with this behaviour?

Reply:
this is gonna be fixed in an upcoming PR - we're gonna roll out the new stuff totally (behind a feature flag) so this intermediate state won't be seen